lib/quicrq_internal.h

```diff
@@ -371,6 +376,64 @@ typedef struct st_quicrq_notify_url_t {
     uint8_t* url;
 } quicrq_notify_url_t;
 
+/* Context representing unidirectional streams*/
+struct st_quicrq_uni_stream_ctx_t {
+    struct st_quicrq_uni_stream_ctx_t* next_stream;
+    struct st_quicrq_uni_stream_ctx_t* previous_stream;
+    struct st_quicrq_cnx_ctx_t* cnx_ctx;
+    /* Source from which data is read and sent on the stream. */
+    quicrq_media_source_ctx_t* media_source;
+    struct st_quicrq_uni_stream_ctx_t* next_stream_for_source;
+    struct st_quicrq_uni_stream_ctx_t* previous_stream_for_source;
+
+    /* Control stream context */
+    struct st_quicrq_stream_ctx_t* control_stream;
+
+    uint64_t stream_id;
+    /* media_id: local identifier of media stream.*/
+    uint64_t media_id;
+    uint64_t next_group_id;
+    uint64_t next_object_id;
+    uint64_t next_object_offset;
+    uint64_t current_group_id;
+    uint64_t start_object_id;
+
```

Comment:
These are properties of the control stream. You don't want to replicate them here. You only need:

 uint64_t uni_stream_id; /* this uni directional stream */
 uint64_t uni_group_id; /* the ID of the group that is sent on this uni stream */
 uint64_t next_uni_object_id; /* The next object to send on this uni stream */